.github/workflows/update-ic-did.yml

```diff
@@ -0,0 +1,65 @@
+name: Update ic.did
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * *" # Runs at UTC midnight every day
+
+env:
+  # When getting Rust dependencies, retry on network error:
+  CARGO_NET_RETRY: 10
+  # Use the local .curlrc
+  CURL_HOME: .
+  # Disable DFX telemetry
+  DFX_TELEMETRY: 'off'
+  # Path to the ic.did file in this repository
+  FILE_PATH: "src/dfx/src/util/ic.did"
+
+jobs:
+  check-update:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout dfx repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Download latest ic.did
+        id: get-file
+        run: |
+          # Download latest version from portal repo
+          EXTERNAL_URL="https://raw.githubusercontent.com/dfinity/portal/refs/heads/master/docs/references/_attachments/ic.did"
+          NEW_CONTENT=$(curl -s $EXTERNAL_URL)
+          # Read the current content of the local file
+          CURRENT_CONTENT=$(cat $FILE_PATH)
+          # Compare contents
+          if [ "$NEW_CONTENT" != "$CURRENT_CONTENT" ]; then
+            echo "has_update=true" >> "$GITHUB_OUTPUT"
+            echo "$NEW_CONTENT" > "$FILE_PATH"
+          else
+            echo "has_update=false" >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
Another option would be `curl -o $FILE_PATH` to download into place, then call `git status --porcelain` which will be empty if there are no changes

Reply:
Applied your suggestion.
The adjusted workflow [succeeded](https://github.com/lwshang/sdk/actions/runs/14363806563).
It created this [PR](https://github.com/lwshang/sdk/pull/2).